cmd/src/debug_kube.go

```diff
@@ -0,0 +1,340 @@
+	flagSet.BoolVar(&noConfigs, "no-configs", false, "If true include Sourcegraph configuration files. Default value true.")
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		// process -o flag to get zipfile and base directory names
+		if base == "" {
+			return fmt.Errorf("empty -o flag")
+		}
+		base, baseDir := processBaseDir(base)
+
+		// init context
+		ctx := context.Background()
+		// open pipe to output file
+		out, err := os.OpenFile(base, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0666)
+		if err != nil {
+			return errors.Wrapf(err, "failed to open file: %w", err)
+		}
+		defer out.Close()
+		// init zip writer
+		zw := zip.NewWriter(out)
+		defer zw.Close()
+
+		// Gather data for safety check
+		pods, err := selectPods(ctx, namespace)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get pods: %w", err)
+		}
+		kubectx, err := exec.CommandContext(ctx, "kubectl", "config", "current-context").CombinedOutput()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get current-context: %w", err)
+		}
+		// Safety check user knows what they've targeted with this command
+		log.Printf("Archiving kubectl data for %d pods\n SRC_ENDPOINT: %v\n Context: %s Namespace: %v\n Output filename: %v", len(pods.Items), cfg.Endpoint, kubectx, namespace, base)
+		if verified, _ := verify("Do you want to start writing to an archive?"); !verified {
+			return nil
+		}
+
+		err = archiveKube(ctx, zw, *verbose, noConfigs, namespace, baseDir, pods)
+		if err != nil {
+			return cmderrors.ExitCode(1, err)
+		}
+		return nil
+	}
+
+	debugCommands = append(debugCommands, &command{
+		flagSet: flagSet,
+		handler: handler,
+		usageFunc: func() {
+			fmt.Println(usage)
+		},
+	})
+}
+
+type podList struct {
+	Items []struct {
+		Metadata struct {
+			Name string
+		}
+		Spec struct {
+			Containers []struct {
+				Name string
+			}
+		}
+	}
+}
+
+// Runs common kubectl functions and archive results to zip file
+func archiveKube(ctx context.Context, zw *zip.Writer, verbose, noConfigs bool, namespace, baseDir string, pods podList) error {
+	// Create a context with a cancel function that we call when returning
+	// from archiveKube. This ensures we close all pending go-routines when returning
+	// early because of an error.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// setup channel for slice of archive function outputs, as well as throttling semaphore
+	ch := make(chan *archiveFile)
+	g, ctx := errgroup.WithContext(ctx)
+	semaphore := semaphore.NewWeighted(8)
+
+	run := func(f func() error) {
+		g.Go(func() error {
+			if err := semaphore.Acquire(ctx, 1); err != nil {
+				return err
+			}
+			defer semaphore.Release(1)
+
+			return f()
+		})
+	}
+
+	// create goroutine to get pods
+	run(func() error {
+		ch <- getPods(ctx, namespace, baseDir)
+		return nil
+	})
```

Comment:
```suggestion
	run(func() *archiveFile { return getPods(ctx, namespace, baseDir) })
```

(example usage of new suggestion)